components/openapi/src/main/scala/pl/touk/nussknacker/openapi/discovery/OpenApiDefinitionDiscovery.scala

```diff
@@ -49,5 +71,20 @@ class SwaggerOpenApiDefinitionDiscovery(implicit val httpBackend: SttpBackend[Fu
     }
     SwaggerParser.parse(definition, openAPIsConfig)
   }
+}
 
+class CachingOpenApiDefinitionDiscovery(
+    discovery: OpenApiDefinitionDiscovery,
+    openAPIsConfig: OpenAPIServicesConfig,
+) extends OpenApiDefinitionDiscovery {
+  @transient private lazy val servicesCache = new SingleValueCache[List[Validated[ServiceParseError, SwaggerService]]](
+    expireAfterAccess = None,
+    expireAfterWrite = Some(openAPIsConfig.openApiServicesDiscoveryCacheTtl)
```

Comment:
Do we need to expire fetched OpenAPI services at all? Currently, the only way to fetch them again is by reloading the model. Maybe we should keep this behaviour unchanged?

Reply:
On the other hand, the cache TTL can be configured using `openApiServicesDiscoveryCacheTtl`, so I think it can be left as it is 🙂

Reply:
> Do we need to expire fetched OpenAPI services at all? Currently, the only way to fetch them again is by reloading the model. Maybe we should keep this behaviour unchanged?
We agreed to change the behaviour because the problem is when nussknacker is starting and openapi service is currently unavailable.

I will take a look at `openApiServicesDiscoveryCacheTtl`

Reply:
That's good point, new behaviour will be consistent with ML components.